src/UserInfoHandler.ts

```diff
@@ -33,11 +33,11 @@ class UserInfo implements LambdaInterface {
 			case ResourcesEnum.USERINFO:
 				if (event.httpMethod === "POST") {
 					try {
-						logger.info("Received userInfo request:", { event });
+						logger.info("Received userInfo request", { event });
```

Comment:
is it safe to log the entire `event` ?

Reply:
Nothing in there stood out to me as PII, but actually there is some API keys. Shall I change to just log `event.headers` maybe?

```{
    "cold_start": true,
    "function_arn": "arn:aws:lambda:eu-west-2:060113405249:function:User-Info-cic-cri-api-ccooling:live",
    "function_memory_size": 1024,
    "function_name": "User-Info-cic-cri-api-ccooling",
    "function_request_id": "e4e863b0-af18-4b16-9ac3-22204cc8a7cf",
    "level": "INFO",
    "message": "Received userInfo request",
    "service": "UserInfoHandler",
    "timestamp": "2023-06-26T13:48:46.950Z",
    "xray_trace_id": "1-6499973e-3ee24292794857cf550c3781",
    "event": {
        "resource": "/userinfo",
        "path": "/userinfo",
        "httpMethod": "POST",
        "headers": {
            "Authorization": "Bearer eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImNlNjk1NzVhLWMzMWItNGM0NS1iNmU1LThkZWE0NjExZDQ1MyJ9.eyJzdWIiOiIzMzgwMzYyNy00ZGMxLTRmMmQtOTUzMy1jODRlMjAyOTc1YmQiLCJhdWQiOiJodHRwczovL3Jldmlldy1jLmRldi5hY2NvdW50Lmdvdi51ayIsImlzcyI6Imh0dHBzOi8vcmV2aWV3LWMuZGV2LmFjY291bnQuZ292LnVrIiwiZXhwIjoxNjg3NzkwOTA1fQ.DkBnds1MVDy4l0LIEQmyo0KxEIA1m77EkdSLY3qTIM0ubJ39suopD8yQvc1Vlyyrg3bbY_dTcRXri8xir7cGkg"
        },
        "multiValueHeaders": {
            "Authorization": [
                "Bearer eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImNlNjk1NzVhLWMzMWItNGM0NS1iNmU1LThkZWE0NjExZDQ1MyJ9.eyJzdWIiOiIzMzgwMzYyNy00ZGMxLTRmMmQtOTUzMy1jODRlMjAyOTc1YmQiLCJhdWQiOiJodHRwczovL3Jldmlldy1jLmRldi5hY2NvdW50Lmdvdi51ayIsImlzcyI6Imh0dHBzOi8vcmV2aWV3LWMuZGV2LmFjY291bnQuZ292LnVrIiwiZXhwIjoxNjg3NzkwOTA1fQ.DkBnds1MVDy4l0LIEQmyo0KxEIA1m77EkdSLY3qTIM0ubJ39suopD8yQvc1Vlyyrg3bbY_dTcRXri8xir7cGkg"
            ]
        },
        "queryStringParameters": null,
        "multiValueQueryStringParameters": null,
        "pathParameters": null,
        "stageVariables": null,
        "requestContext": {
            "resourceId": "5pirf5",
            "resourcePath": "/userinfo",
            "operationName": "postUserInfo",
            "httpMethod": "POST",
            "extendedRequestId": "HISRvEZCLPEFW-w=",
            "requestTime": "26/Jun/2023:13:48:46 +0000",
            "path": "/userinfo",
            "accountId": "060113405249",
            "protocol": "HTTP/1.1",
            "stage": "test-invoke-stage",
            "domainPrefix": "testPrefix",
            "requestTimeEpoch": 1687787326055,
            "requestId": "cef25f75-7a57-4bc9-a24f-1a25c76074bc",
            "identity": {
                "cognitoIdentityPoolId": null,
                "cognitoIdentityId": null,
                "apiKey": "test-invoke-api-key",
                "principalOrgId": null,
                "cognitoAuthenticationType": null,
                "userArn": "arn:aws:sts::060113405249:assumed-role/caitlin.cooling-admin/1687785995978872000",
                "apiKeyId": "test-invoke-api-key-id",
                "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36",
                "accountId": "060113405249",
                "caller": "AROAQ37YJYVASRDQNXHUO:1687785995978872000",
                "sourceIp": "test-invoke-source-ip",
                "accessKey": "ASIAQ37YJYVA3OSG5PGJ",
                "cognitoAuthenticationProvider": null,
                "user": "AROAQ37YJYVASRDQNXHUO:1687785995978872000"
            },
            "domainName": "testPrefix.testDomainName",
            "apiId": "aj43ht3kx2"
        },
        "body": null,
        "isBase64Encoded": false
    }
}
```

Reply:
Its that `Authorization` header value which contains `subjectId` PII, lets not log the `event` object

Reply:
Have discussed with Darren, while the sub is fine to log, it is not fine to log a live bearer token. I have changed it so that we are now only logging the api gateway request ID 👍